homeassistant/components/sensor/bmw_connected_drive.py

```diff
@@ -0,0 +1,89 @@
+"""
+Reads vehicle status from BMW connected drive portal.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.bmw_connected_drive/
+"""
+import logging
+
+from homeassistant.components.bmw_connected_drive import DOMAIN as BMW_DOMAIN
+from homeassistant.helpers.entity import Entity
+
+
+_LOGGER = logging.getLogger(__name__)
+
+LENGTH_ATTRIBUTES = [
+    'remaining_range_fuel',
+    'mileage',
+    ]
+
+VAILD_ATTRIBUTES = LENGTH_ATTRIBUTES + [
+    'remaining_fuel',
+]
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the BMW sensors."""
+    entities = hass.data[BMW_DOMAIN]
+    _LOGGER.debug('Found BMW accounts: %s',
+                  ', '.join([v.name for v in entities]))
+    devices = []
+    for entity in entities:
+        account = entity.account
+        for vehicle in account.vehicles:
+            for sensor in VAILD_ATTRIBUTES:
+                device = BMWConnectedDriveSensor(entity, account,
+                                                 vehicle, sensor)
+                devices.append(device)
+    return add_devices(devices)
+
+
+class BMWConnectedDriveSensor(Entity):
+    """Representation of a BMW vehicle."""
+
+    def __init__(self, entity, account, vehicle, attribute: str):
+        """Constructor."""
+        self._vehicle = vehicle
+        self._attribute = attribute
+        self._state = None
+        self._unit_of_measurement = None
+        self._name = '{} {}'.format(self._vehicle.modelName, self._attribute)
+        entity.add_update_listener(self.update)
```

Comment:
Move the registration of the callback to `async_added_to_hass`. Otherwise the callback might be called before the entity is added to home assistant which will error.

Reply:
done